packages/edit-site/src/components/page-patterns/index.js

```diff
@@ -116,14 +119,21 @@ function Preview( { item, viewType } ) {
 	const descriptionId = useId();
 	const isUserPattern = item.type === PATTERN_TYPES.user;
 	const isTemplatePart = item.type === TEMPLATE_PART_POST_TYPE;
-	const isEmpty = ! item.blocks?.length;
-
 	const [ backgroundColor ] = useGlobalStyle( 'color.background' );
 	const { onClick } = useLink( {
 		postType: item.type,
-		postId: isUserPattern ? item.id : item.name,
+		postId: isUserPattern || isTemplatePart ? item.id : item.name,
 		canvas: 'edit',
 	} );
+	const blocks = useMemo( () => {
+		return (
+			item.blocks ??
```

Comment:
Being more intentional about providing blocks to preview sounds good, but I'm not sure _when_ the `item.blocks` could be empty. Have you run into this or have a use case I test?

Reply:
item.blocks doesn't exist in template parts by default, it only exists if you get the edited template part objects after it was modified in the editor (and not saved)

Reply:
I was checking that now. This was something that we did before on `templatePartToPattern`. 👍